vision_evaluation/evaluators.py

```diff
@@ -1046,3 +1046,64 @@ def add_predictions(self, predictions, targets):
             gt_mask = np.asarray(gt_mask)
             mean_l1 = np.abs(pred_mask.astype(np.float)-gt_mask.astype(np.float)).mean()
             self._metric_sum += mean_l1
+
+
+
+class GroupWiseEvaluator(Evaluator):
+    """
+    Group Wise Evaluator
+    """
+
+    def __init__(self, evaluator):
+        super().__init__()
+        self.evaluator = evaluator
+
+    def _group_by(self):
+        """
+            Group predictions and targets by target_task and group labels.
+            Example: IC on target_task = 'ethnicity' and group = 'gender'
+
+            store dict will save target labels and predictions corresponding to each group label.
+
+            store = {(target_task, group) : {
+                "targets":[]
+                "predictions":[]
+            }}
+        """
+
+        self.store = collections.defaultdict(
+            lambda: collections.defaultdict(list))
+
+        for (target_task, group), prediction in zip(self.targets, self.predictions):
+            self.store[(target_task, group)]['predictions'].append(prediction)
+            self.store[(target_task, group)]['targets'].append(target_task)
```

Comment:
why need the 'targets' key, given the elements stored in 'targets' are always `target_task`?

Reply:
for consistency?

Reply:
sorry, could you elaborate more on the consistency part?